src/context/WalletProvider/GridPlus/components/SafeCardList.tsx

```diff
@@ -0,0 +1,325 @@
+            borderWidth={1}
+            borderRadius='lg'
+            borderColor='border.base'
+            _hover={hoverSx}
+            bg='background.surface.raised.base'
+          >
+            <HStack spacing={3}>
+              <Box color='text.subtle'>
+                <FaWallet size={20} />
+              </Box>
+
+              {editingId === safeCard.id ? (
+                <Input
+                  size='sm'
+                  value={editName}
+                  onChange={handleEditNameChange}
+                  data-id={safeCard.id}
+                  onBlur={handleRenameBlur}
+                  onKeyDown={handleRenameKeyDown}
+                  autoFocus
+                  flex={1}
+                />
+              ) : (
+                <VStack align='start' flex={1} spacing={0}>
+                  <Text fontWeight='medium'>{safeCard.name}</Text>
+                  {safeCard.lastConnectedAt && (
+                    <Text fontSize='xs' color='text.subtle'>
+                      {translate('walletProvider.gridplus.list.lastConnected', {
+                        date: new Date(safeCard.lastConnectedAt).toLocaleDateString(),
+                      })}
+                    </Text>
+                  )}
+                </VStack>
+              )}
+
+              <ButtonGroup size='sm'>
+                {editingId !== safeCard.id && (
+                  <IconButton
+                    icon={CreateIcon}
+                    aria-label={translate('walletProvider.gridplus.list.editNameLabel')}
+                    variant='ghost'
+                    data-id={safeCard.id}
+                    data-name={safeCard.name}
+                    onClick={handleEditClick}
+                  />
+                )}
+                <IconButton
+                  icon={TrashIcon}
+                  aria-label={translate('walletProvider.gridplus.list.deleteLabel')}
+                  variant='ghost'
+                  colorScheme='red'
+                  data-id={safeCard.id}
+                  onClick={handleDeleteClick}
+                />
+                <Button
+                  data-id={safeCard.id}
+                  onClick={handleSelectClick}
+                  colorScheme='blue'
+                  isDisabled={isActive || isConnecting}
+                  isLoading={isConnecting}
+                >
+                  {isActive
+                    ? translate('walletProvider.gridplus.list.connected')
+                    : translate('walletProvider.gridplus.list.connect')}
+                </Button>
+              </ButtonGroup>
+            </HStack>
+          </Box>
+        )
+      }),
+    [
+      sortedSafeCards,
+      walletState.isConnected,
+      walletState.deviceId,
+      connectingCardId,
+      editingId,
+      editName,
+      handleEditNameChange,
+      handleRenameBlur,
+      handleRenameKeyDown,
+      handleEditClick,
+      handleDeleteClick,
+      handleSelectClick,
+      translate,
+    ],
+  )
```

Comment:
We could probably extract that and a Card component to simplify things a little bit